pkg/ansible/runner.go

```diff
@@ -0,0 +1,208 @@
+		},
+		{
+			Name: "AWS_ACCESS_KEY_ID",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_access_key_id",
+				},
+			},
+		},
+		{
+			Name: "AWS_SECRET_ACCESS_KEY",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_secret_access_key",
+				},
+			},
+		},
+	}
+	runAsUser := int64(0)
+	sshKeyFileMode := int32(0600)
+	playbookPath := path.Join(openshiftAnsibleContainerDir, playbook)
+	podSpec := kapi.PodSpec{
+		DNSPolicy:          kapi.DNSClusterFirst,
+		RestartPolicy:      kapi.RestartPolicyNever,
+		ServiceAccountName: openshiftAnsibleServiceAccount,
+
+		Containers: []kapi.Container{
+			{
+				Name:  "ansible",
+				Image: r.Image,
+				Env:   env,
+				SecurityContext: &kapi.SecurityContext{
+					RunAsUser: &runAsUser,
+				},
+				VolumeMounts: []kapi.VolumeMount{
+					{
+						Name:      "inventory",
+						MountPath: "/ansible/inventory/",
+					},
+					{
+						Name:      "sshkey",
+						MountPath: "/ansible/ssh/",
+					},
+				},
+
+				// TODO: possibly drop this once https://github.com/openshift/openshift-ansible/pull/6320 merges, the default run script should then work, minus the vars file :(
+				//Command: []string{"ansible-playbook", "-i", "/ansible/inventory/hosts", "--private-key", "/ansible/ssh/privatekey.pem", playbookPath, "-e", "@/ansible/inventory/vars"},
+
+				// TODO: committing as no-op for now, uncomment command above to start testing provisioning
+				Command: []string{"echo", playbookPath},
+			},
+		},
+		Volumes: []kapi.Volume{
+			{
+				// Mounts both our inventory and vars file.
+				Name: "inventory",
+				VolumeSource: kapi.VolumeSource{
+					ConfigMap: &kapi.ConfigMapVolumeSource{
+						LocalObjectReference: kapi.LocalObjectReference{
+							Name: cfgMap.Name,
+						},
+					},
+				},
+			},
+			{
+				Name: "sshkey",
+				VolumeSource: kapi.VolumeSource{
+					Secret: &kapi.SecretVolumeSource{
+						SecretName: sshPrivateKeySecret,
+						Items: []kapi.KeyToPath{
+							{
+								Key:  "ssh-privatekey",
+								Path: "privatekey.pem",
+								Mode: &sshKeyFileMode,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	completions := int32(1)
+	deadline := int64(60 * 60) // one hour for now
+
+	// TODO: right now we get a configmap names ansible-job-cluster-X, and a job
+	// named ansible-job-cluster-Y. Find a good way to link these two randomly named
+	// resources. Note that jobs implicitly seem to get a random prefix, even if you
+	// try to set the name directly.
+	genNamePrefix := fmt.Sprintf("%s%s-", namePrefix, clusterName)
+	meta := metav1.ObjectMeta{
+		GenerateName: genNamePrefix,
+		Namespace:    operatorNamespace,
```

Comment:
It is not necessary to set the namespace here. The namespace will be set by the API server.